modules/catalog/src/main/java/org/apache/ignite/internal/catalog/commands/AlterZoneCommand.java

```diff
@@ -136,23 +148,51 @@ private CatalogZoneDescriptor fromParamsAndPreviousValue(CatalogZoneDescriptor p
         CatalogStorageProfilesDescriptor storageProfiles = storageProfileParams != null
                 ? fromParams(storageProfileParams) : previous.storageProfiles();
 
+        int replicas = requireNonNullElse(this.replicas, previous.replicas());
+        int quorumSize = adjustQuorumSize(replicas, previous.quorumSize());
+
         return new CatalogZoneDescriptor(
                 previous.id(),
                 previous.name(),
-                Objects.requireNonNullElse(partitions, previous.partitions()),
-                Objects.requireNonNullElse(replicas, previous.replicas()),
-                Objects.requireNonNullElse(autoAdjust, previous.dataNodesAutoAdjust()),
-                Objects.requireNonNullElse(scaleUp, previous.dataNodesAutoAdjustScaleUp()),
-                Objects.requireNonNullElse(scaleDown, previous.dataNodesAutoAdjustScaleDown()),
-                Objects.requireNonNullElse(filter, previous.filter()),
+                requireNonNullElse(partitions, previous.partitions()),
+                replicas,
+                quorumSize,
+                requireNonNullElse(autoAdjust, previous.dataNodesAutoAdjust()),
+                requireNonNullElse(scaleUp, previous.dataNodesAutoAdjustScaleUp()),
+                requireNonNullElse(scaleDown, previous.dataNodesAutoAdjustScaleDown()),
+                requireNonNullElse(filter, previous.filter()),
                 storageProfiles,
                 previous.consistencyMode()
         );
     }
 
+    /**
+     * Adjusts quorum size so it is always consistent with the replicas count.
+     *
+     * @param replicas Desired replicas count.
+     * @param previousQuorumSize Previous quorum size.
+     * @return Adjusted quorum size.
+     */
+    private int adjustQuorumSize(int replicas, int previousQuorumSize) {
+        int quorumSize = requireNonNullElse(this.quorumSize, previousQuorumSize);
+
+        int minQuorum = min(replicas, 2);
+        int maxQuorum = max(minQuorum, (int) (round(replicas / 2.0)));
+
+        if (quorumSize > maxQuorum) {
+            LOG.info("Quorum size adjusted from {} to {}.", quorumSize, maxQuorum);
+            return maxQuorum;
+        } else if (quorumSize < minQuorum) {
+            LOG.info("Quorum size adjusted from {} to {}.", quorumSize, minQuorum);
```

Comment:
```suggestion
            LOG.info("Quorum size adjusted from {} to {} because it is less than the minimum quorum value.", quorumSize, minQuorum);
```